source/fe/mapping_q_generic.cc

```diff
@@ -106,7 +106,7 @@ namespace internal
         const double discriminant = b*b - 4*a*c;
         // exit if the point is not in the cell (this is the only case where the
         // discriminant is negative)
-        if (discriminant < 0.0)
+        if (discriminant < 0.0 || ( a == 0.0 && b == 0.0))
```

Comment:
I have opened #4450 where this one is addressed. Indeed the case I wanted to catch was `a==0` and `b==0` in which case the discriminant is always zero, so your suggestion works just fine.